Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -1147,9 +1155,10 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
         try? tokenStore.deleteToken()
 #endif
 
-        // This is not really an error, we received a command to reset the connection
-        cancelTunnelWithError(nil)
-        completionHandler?(nil)
+        Task {
+            completionHandler?(nil)
+            await cancelTunnel(with: TunnelError.appRequestedCancellation)
```

Comment:
Any reason the completion handler comes before the cancel operation?

Reply:
Yeah - I want to make sure the completion handler is always called to prevent things from getting into a weird state.  It's very likely that cancelling the tunnel will try to deallocate the `PacketTunnelProvider` and I'm ware of what might happen if the completion wasn't called before.